### PR TITLE
feat: Add getPartialCircleDimensions to Utility

### PR DESCRIPTION
**Pull request by jbrundage**
Add degreesToRadians to Utility
Add radiansToDegrees to Utility

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

### CODE REVIEW

**Comment by jbrundage** [comment]
@GordonSmith `Utility.getPartialCircleDimensions` can be used to optimize the placement of the `HalfDonut` and `QuarterDonut` widgets without needing to get the bbox from the DOM.

It returns the bounding height and width of a partial circle:
image

File: packages/common/src/Utility.ts [hunk 1]
```diff
@@ -587,3 +587,36 @@ export function removeHTMLFromString(str: string, div?: HTMLDivElement) {
...
+    if (firstAngle <= Math.PI * 1.0 && lastAngle >= Math.PI * 1.0) point_arr.push([-radius,       0]);
+    if (firstAngle <= Math.PI * 1.5 && lastAngle >= Math.PI * 1.5) point_arr.push([      0, -radius]);
+    if (firstAngle <= Math.PI * 2.0 && lastAngle >= Math.PI * 2.0) point_arr.push([ radius,       0]);
+    point_arr.push([x1, y1]);
+    point_arr.push([x2, y2]);
+    point_arr.push([0, 0]);
+    const minX = Math.min.apply(undefined, point_arr.map(n => n[0]));
+    const minY = Math.min.apply(undefined, point_arr.map(n => n[1]));
+    const maxX = Math.max.apply(undefined, point_arr.map(n => n[0]));
+    const maxY = Math.max.apply(undefined, point_arr.map(n => n[1]));
```
Comment (GordonSmith) [inline packages/common/src/Utility.ts]: Would be a little cleaner to use https://github.com/d3/d3-array#extent

File: packages/common/src/Utility.ts [hunk 2]
```diff
@@ -587,3 +587,36 @@ export function removeHTMLFromString(str: string, div?: HTMLDivElement) {
...
+    return {
+        width: maxX - minX,
+        height: maxY - minY
+    };
+}
```
Comment (GordonSmith) [inline packages/common/src/Utility.ts]: I think these should be in @hpcc-js/util in a file called geometry?
Also in util, you will see everything has unit tests and a lot of stuff has some API documentation - which is probably a good idea for these types of utility?

**Comment by GordonSmith** [comment]
Quick question - the center point (shown in the image) is not used as part of the bounding box equation correct?

**Comment by jbrundage** [comment]
The center point is used as part of the bounding box equation. Example: 
If start angle is 30 and end angle is 60 we need the center point to represent the top and left bounds.
image

**Comment by GordonSmith** [comment]
I think I would rename it to:  getPartialPieDimensions so?

**Comment by GordonSmith** [comment]
Or Partial Disc ?

File: packages/util/src/geometry.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    if (firstAngle <= Math.PI * 0.5 && lastAngle >= Math.PI * 0.5) point_arr.push([      0,  radius]);
+    if (firstAngle <= Math.PI * 1.0 && lastAngle >= Math.PI * 1.0) point_arr.push([-radius,       0]);
+    if (firstAngle <= Math.PI * 1.5 && lastAngle >= Math.PI * 1.5) point_arr.push([      0, -radius]);
+    if (firstAngle <= Math.PI * 2.0 && lastAngle >= Math.PI * 2.0) point_arr.push([ radius,       0]);
+    point_arr.push([x1, y1]);
+    point_arr.push([x2, y2]);
+    point_arr.push([0, 0]);
+    const x_arr = point_arr.map(n => n[0]);
+    const y_arr = point_arr.map(n => n[1]);
+    const minX = Math.min.apply(undefined, x_arr);
```
Comment (GordonSmith) [inline packages/util/src/geometry.ts]: `const minX = Math.min(...x_arr);` would be cleaner

**Comment by jbrundage** [comment]
I'm not sure what happened to it... I don't see it in my stash list... investigating now

**Comment by GordonSmith** [comment]
@jbrundage bump

**Comment by GordonSmith** [comment]
@jbrundage can you rebase and I will merge.

**Comment by GordonSmith** [comment]
@jbrundage  bump

**Pull request closed without merging**